core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/LoadSequencerWorkflow.java

```diff
@@ -24,47 +44,282 @@ protected WorkflowName getWorkflowName() {
   }
 
   @Override
-  public WorkflowStepPrompt getNextStep() {
-    return null;
+  public WorkflowStepPrompt getStep(int stepNumber) {
+    switch (stepNumber) {
+    case 0:
+      return containerStep.getPrompt();
+    case 1:
+      return containerStep.isExistingContainer() ? partitionSteps.get(0).getPrompt() : containerModelStep.getPrompt();
+    default:
+      return partitionSteps.get(asPartitionIndex(stepNumber)).getPrompt();
+    }
   }
 
-  @Override
-  public WorkflowStepPrompt getStep(int stepNumber) {
-    return null;
+  private int asPartitionIndex(int stepNumber) {
+    return stepNumber - (containerStep.isExistingContainer() ? 1 : 2);
   }
 
   @Override
   public boolean isComplete() {
-    return false;
+    return partitionSteps.size() > 0 && Stream.concat(Stream.of(containerStep), partitionSteps.stream()).allMatch(this::hasInput);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)